mypy/semanal_enum.py

```diff
@@ -106,10 +106,23 @@ def parse_enum_call_args(self, call: CallExpr,
         args = call.args
         if len(args) < 2:
             return self.fail_enum_call_arg("Too few arguments for %s()" % class_name, call)
-        if len(args) > 2:
-            return self.fail_enum_call_arg("Too many arguments for %s()" % class_name, call)
-        if call.arg_kinds != [ARG_POS, ARG_POS]:
-            return self.fail_enum_call_arg("Unexpected arguments to %s()" % class_name, call)
+        elif len(args) == 2:
+            if call.arg_kinds != [ARG_POS, ARG_POS]:
+                return self.fail_enum_call_arg("Unexpected arguments to %s()" % class_name, call)
+        elif len(args) > 2:
+            if call.arg_kinds[:2] != [ARG_POS, ARG_POS]:
+                return self.fail_enum_call_arg("Too many arguments for %s()" % class_name, call)
+            if all(arg == ARG_NAMED_OPT for arg in call.arg_kinds[2:]):
+                return self.fail_enum_call_arg("Too many arguments for %s()" % class_name, call)
+
+            for arg in call.arg_names:
+                if arg is None:
```

Comment:
I'm confused about this case? Won't this fire because of the positional arguments?

Reply:
I am worried about that too. Should I just not check for the first two elements of the list `arg_names`?